terraform/modules/front_door/waf.tf

```diff
@@ -0,0 +1,164 @@
+provider "aws" {
+  alias  = "north-virginia"
+  region = "us-east-1"
+
+  assume_role {
+    role_arn = var.provider_role_arn
+  }
+}
+
+resource "aws_wafv2_web_acl" "this" {
+  #checkov:skip=CKV2_AWS_31:TODO CLDC-2781 setup WAF logging in cloudwatch
+  name        = var.prefix
+  description = "Web ACL to restrict traffic to CloudFront"
+  provider    = aws.north-virginia
+  scope       = "CLOUDFRONT"
+
+  default_action {
+    allow {}
+  }
+
+  visibility_config {
+    cloudwatch_metrics_enabled = true
+    metric_name                = "waf-metrics"
+    sampled_requests_enabled   = false
+  }
+
+  rule {
+    name     = "AWSManagedRulesAmazonIpReputationList"
+    priority = 1
+
+    override_action {
+      none {}
+    }
+
+    statement {
+      managed_rule_group_statement {
+        name        = "AWSManagedRulesAmazonIpReputationList"
+        vendor_name = "AWS"
+      }
+    }
+
+    visibility_config {
+      cloudwatch_metrics_enabled = true
+      metric_name                = "waf-block-disreputable-ip-metrics"

```

Comment:
Minor point - do we perhaps want to suffix individual metrics with `metric` rather than `metrics`? Or do we even need a suffix at all - might it not be obvious what it is when looking in the console?

Reply:
Yeah I think this was added in EPB to make the code more readable and I did something similar here, but in the console it's probably just noise (see screenshot), so I'll remove it 👍 

![image](https://github.com/communitiesuk/submit-social-housing-lettings-and-sales-data-infrastructure/assets/109098469/d201004c-f1e6-4768-8d9d-a398bb8cc0bf)

Reply:
Removed 👍